src/dragonball/src/vmm.rs

```diff
@@ -137,7 +139,13 @@ impl Vmm {
                     }
                     if v.event_ctx.exit_evt_triggered {
                         info!("Gracefully terminated VMM control loop");
-                        return v.stop(EXIT_CODE_OK as i32);
+                        let ret = v.stop(EXIT_CODE_OK as i32);
+                        let tracer_lock = service.tracer();
+                        let mut tracer = tracer_lock.lock().unwrap();
+                        if tracer.enabled() {
+                            tracer.end_tracing().expect("End tracing err");
```

Comment:
tracer.enabled is checked in end_tracing()

Reply:
Removed.